src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java

```diff
@@ -110,6 +121,14 @@ public StorageAdaptorRetrieveResponse retrieve(@NotBlank final String key)
     }
   }
 
+  @Override
+  public void updateStatus(String datasetId, String storeStatus) {
```

Comment:
I think we should protect against setting any type of status, because only a few should be allowed.

Reply:
That did cross my mind. I'll look into changing it.